### ISSUE TITLE
Unable to search packages without user & channel

### ISSUE THREAD

**Issue by danimtb** [conan-io/conan#5644]
Searching packages that have been created without user and channel is not possible:

```
$ conan search kk*
Existing package recipes:

kk/1.0
kk/1.0@name/version
kk/1.0@us/ch

$ conan search kk/1.0
Existing package recipes:

kk/1.0
kk/1.0@name/version
kk/1.0@us/ch
```

looking at the results, I think the ``kk/1.0`` is managed as a pattern.

Ok, this actually works, but I think it is not what the user expects:

```
$ conan search kk/1.0@
Existing packages for recipe kk/1.0:

    Package_ID: 6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7
        [options]
            shared: False
        [settings]
            arch: x86_64
            build_type: Release
            compiler: Visual Studio
            compiler.runtime: MD
            compiler.version: 15
            os: Windows
        Outdated from recipe: True
```

**Comment by lasote** [comment]
`kk/1.0@` is a valid reference. The same as `kk/1.0@user/channel`.
At Conan 2.0, another thing to disambiguate is when are we using a pattern and when are we using a reference.

**Comment by memsharded** [comment]
Already fixed in Conan 2.0, available in 2.0-alpha1 with the explicit ``conan list`` command.
